src/operator/inspect.rs

```diff
@@ -76,12 +77,17 @@ where
     }
 }
 
-impl<T, F> SinkOperator<T> for Inspect<T, F>
+impl<T, F> UnaryOperator<T, T> for Inspect<T, F>
 where
-    T: 'static,
+    T: Clone + 'static,
     F: FnMut(&T) + 'static,
 {
-    fn eval(&mut self, i: &T) {
-        (self.callback)(i)
+    fn eval(&mut self, i: &T) -> T {
+        (self.callback)(i);
+        i.clone()
+    }
+    fn eval_owned(&mut self, i: T) -> T {
```

Comment:
Is this output used for anything? Can this just return ()?

Reply:
It's used by the next operator in the chain.